crates/primitives/src/transaction/mod.rs

```diff
@@ -820,6 +820,188 @@ impl Encodable for Transaction {
     }
 }
 
+impl alloy_consensus::Transaction for Transaction {
+    fn chain_id(&self) -> Option<ChainId> {
+        match self {
+            Self::Legacy(tx) => tx.chain_id(),
+            Self::Eip2930(tx) => tx.chain_id(),
+            Self::Eip1559(tx) => tx.chain_id(),
+            Self::Eip4844(tx) => tx.chain_id(),
+            Self::Eip7702(tx) => tx.chain_id(),
+            #[cfg(feature = "optimism")]
+            Self::Deposit(tx) => tx.chain_id(),
+        }
+    }
```

Comment:
@mattsse this means also that we can, in a follow up PR, remove the redundant function definitions because a `chain_id()` method already existed for `Transaction` as part of its direct implementation but with this trait, this is not needed anymore. This is also true for other methods in the list.